server_cmds.go

```diff
@@ -15,129 +15,149 @@ type ServerMethods struct {
+	PacketsSentKeepalive                   uint64        `ms:"connection_packets_sent_keepalive"`
+	BytesSentKeepalive                     uint64        `ms:"connection_bytes_sent_keepalive"`
+	PacketsReceivedKeepalive               uint64        `ms:"connection_packets_received_keepalive"`
+	BytesReceivedKeepalive                 uint64        `ms:"connection_bytes_received_keepalive"`
+	PacketsSentControl                     uint64        `ms:"connection_packets_sent_control"`
+	BytesSentControl                       uint64        `ms:"connection_bytes_sent_control"`
+	PacketsReceivedControl                 uint64        `ms:"connection_packets_received_control"`
+	BytesReceivedControl                   uint64        `ms:"connection_bytes_received_control"`
+	PacketsSentTotal                       uint64        `ms:"connection_packets_sent_total"`
+	BytesSentTotal                         uint64        `ms:"connection_bytes_sent_total"`
+	PacketsReceivedTotal                   uint64        `ms:"connection_packets_received_total"`
+	BytesReceivedTotal                     uint64        `ms:"connection_bytes_received_total"`
+	BandwidthSentLastSecondTotal           uint64        `ms:"virtualserver_bandwidth_sent_last_second_total"`
+	BandwidthSentLastMinuteTotal           uint64        `ms:"virtualserver_bandwidth_sent_last_minute_total"`
+	BandwidthReceivedLastSecondTotal       uint64        `ms:"virtualserver_bandwidth_received_last_second_total"`
+	BandwidthReceivedLastMinuteTotal       uint64        `ms:"virtualserver_bandwidth_received_last_minute_total"`
+	MonthBytesDownloaded                   uint64        `ms:"virtualserver_month_bytes_downloaded"`
+	MonthBytesUploaded                     uint64        `ms:"virtualserver_month_bytes_uploaded"`
+	TotalBytesDownloaded                   uint64        `ms:"virtualserver_total_bytes_downloaded"`
+	TotalBytesUploaded                     uint64        `ms:"virtualserver_total_bytes_uploaded"`
+	TotalPacketLossSpeech                  float64       `ms:"virtualserver_total_packetloss_speech"`
+	TotalPacketLossKeepalive               float64       `ms:"virtualserver_total_packetloss_keepalive"`
+	TotalPacketLossControl                 float64       `ms:"virtualserver_total_packetloss_control"`
+	TotalPacketLossTotal                   float64       `ms:"virtualserver_total_packetloss_total"`
+	TotalPing                              float32       `ms:"virtualserver_total_ping"`
+	PrioritySpeakerDimmModificator         float32       `ms:"virtualserver_priority_speaker_dimm_modificator"`
+	MaxClients                             uint16        `ms:"virtualserver_maxclients"`
+	ClientsOnline                          uint16        `ms:"virtualserver_clientsonline"`
+	ChannelsOnline                         uint16        `ms:"virtualserver_channelsonline"`
+	Uptime                                 time.Duration `ms:"virtualserver_uptime"`
```

Comment:
I don't see any custom unmarshaling or fix up so I think this will have the wrong value as uptime is reported in seconds not nano seconds, same for the other time.Duration fields.

Reply:
I am fairly new to Go, so would you mind to point me into a direction to fix this?

Reply:
I am unsure if this method is "the" correct way to do this and I would love some feedback on this.
The good news is, it does work correctly.

Reply:
This is a good question and truth be told there's no perfect way of doing it for a json unmarshal, but a good article which goes over the options is here:
https://blog.gopheracademy.com/advent-2016/advanced-encoding-decoding/

For ours however we're using `mapstructure` which has a custom decoding support.

I think we could extend the hook which is here:
https://github.com/multiplay/go-ts3/blob/master/helpers.go#L150